src/connectors/hits-per-page/connectHitsPerPage.js

```diff
@@ -167,10 +170,16 @@ Learn more: https://community.algolia.com/instantsearch.js/v2/widgets/configure.
             ? helper.setQueryParameter('hitsPerPage', undefined).search()
             : helper.setQueryParameter('hitsPerPage', value).search();
 
+        createURLFn = value => {
+          this.setHitsPerPage(value);
```

Comment:
The `createURL` function should not set the page because otherwise it will trigger a search every time it's called. The function must generate the URL that matches the provided value. You can take a look at the implementation inside [`connectMenu`](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/menu/connectMenu.js#L182-L183).

Reply:
Right, it would work as expected if we don't call `.search()`, right? If not, what helper function should it use then?

Reply:
We don't have to call methods on the helper directly (because it mutates the state on it). What we have to do is (what is done inside [`connectMenu`](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/menu/connectMenu.js#L182-L183)) call the method on the `state` attribute. It won't mutate the internal helper state.

Reply:
Right, I see!